cpp/velox/shuffle/VeloxShuffleWriter.h

```diff
@@ -132,6 +132,8 @@ class VeloxShuffleWriter final : public ShuffleWriter {
 
   arrow::Status split(std::shared_ptr<ColumnarBatch> cb, int64_t memLimit) override;
 
+  arrow::Status sort(std::shared_ptr<ColumnarBatch> cb, int64_t memLimit) override;
```

Comment:
We may seperate the sort based and hash based shuffle in different files

Reply:
@FelixYBW After Sort's proposal is reviewed, I'll refactor the code structure to meet the requirements of the different files.

Reply:
@kerwin-zk After this patch, part of the existing APIs are only used for hash-based shuffle, and the newly added ones are only used for sort-based shuffle, which leads to a mixing of the code. It would be nice to separate the APIs for both writer & reader.

Reply:
@marin-ma  Shuffle Writer's api has been split into sort and hash, for Shuffle Reader, it just adds Velox Stream parsing, and doesn't differentiate between sort and hash, the core logic is the same, I don't feel the need to split it out separately.